OpenRA.Game/Widgets/WidgetLoader.cs

```diff
@@ -48,6 +48,15 @@ public Widget LoadWidget(WidgetArgs args, Widget parent, MiniYamlNode node)
 			if (!args.ContainsKey("modRules"))
 				args = new WidgetArgs(args) { { "modRules", modData.DefaultRules } };
 
+			object parentLogic = null;
+			if (args.ContainsKey("logic"))
+			{
+				parentLogic = args["logic"];
+				args.Remove("logic");
+			}
+
+			args.Add("logic", node.Value.Nodes.FirstOrDefault(n => n.Key == "Logic"));
```

Comment:
In this case we should take only the last value.  You can either do it manually here or push it into an overload of ToDictionary - this is something that will be useful in other places too.